### PR TITLE
Documentation for `infleqtion` target

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Follow-up to https://github.com/NVIDIA/cuda-quantum/pull/2423 by adding documentation on the `infleqtion` target.

This PR additionally includes an end-to-end application workflow demo notebook using CUDA-Q to execute on the `infleqtion` target

### CODE REVIEW

File: docs/sphinx/targets/cpp/infleqtion.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
```
Comment (khalatepradnya) [inline docs/sphinx/targets/cpp/infleqtion.cpp:1]: The examples don't have copyright banner. Please remove for consistency.

File: docs/sphinx/targets/python/infleqtion.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# ============================================================================ #
```
Comment (khalatepradnya) [inline docs/sphinx/targets/python/infleqtion.py:1]: Same here.

File: docs/sphinx/using/backends/hardware.rst [hunk 3]
```diff
@@ -5,6 +5,129 @@ CUDA-Q supports submission to a set of hardware providers.
 To submit to a hardware backend, you need an account with the respective provider.
 
 
+Infleqtion
```
Comment (khalatepradnya) [inline docs/sphinx/using/backends/hardware.rst:8]: These sections are in alphabetical order.

File: docs/sphinx/using/backends/hardware.rst [hunk 4]
```diff
@@ -5,6 +5,129 @@ CUDA-Q supports submission to a set of hardware providers.
...
+
+    cudaq.set_target("infleqtion", machine="cq_sqale_qpu")
+
+where ``cq_sqale_qpu`` is an example of a physical QPU.
+
+To run an ideal dry-run execution of the QPU, additionally set the ``method`` flag to ``"dry-run"``.
+
+.. code:: python
+
+    cudaq.set_target('infleqtion', machine="cq_sqale_qpu", method="dry-run")
```
Comment (khalatepradnya) [inline docs/sphinx/using/backends/hardware.rst:102]: nit: Use double quotes throughout for consistency.
```cudaq.set_target("infleqtion", machine="cq_sqale_qpu", method="dry-run")```

File: docs/sphinx/using/backends/hardware.rst (hunk 3 above)
Comment (bharat-thotakura) [reply docs/sphinx/using/backends/hardware.rst:115]: Apologies, reordered in https://github.com/NVIDIA/cuda-quantum/pull/2464/commits/75a0a08721ec16d202ef399e295c7ff82b7f9136

File: docs/sphinx/using/backends/hardware.rst (hunk 4 above)
Comment (bharat-thotakura) [reply docs/sphinx/using/backends/hardware.rst]: Thanks for the catch, fixed now

File: docs/sphinx/using/backends/hardware.rst [hunk 5]
```diff
@@ -112,6 +112,128 @@ To see a complete example for using Amazon Braket backends, take a look at our :
...
+the ``shots_count`` argument to ``cudaq.sample`` or ``cudaq.observe``. By default,
+the ``shots_count`` is set to 1000.
+
+.. code:: python
+
+    cudaq.sample(kernel, shots_count=100)
+
+To see a complete example for using Infleqtion's backends, take a look at our :doc:`Python examples <../examples/examples>`.
+Moreover, for an end-to-end application workflow example executed on the Infleqtion QPU, take a look at the 
+:doc:`AIM ground state solver <../applications/python/logical_aim_sqale>` notebook.
```
Comment (bettinaheim) [inline docs/sphinx/using/backends/hardware.rst]: This link doesn't work as is. I suggest to instead link to the applications page, and list teh new AIM ground state solver there.
```suggestion
:doc:`AIM ground state solver <../applications>`.
```

Comment (bettinaheim) [reply docs/sphinx/using/backends/hardware.rst]: I took the liberty of pushing a commit that does that.

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**